kalite/packages/bundled/fle_utils/django_utils/command.py

```diff
@@ -118,6 +118,73 @@ def call_command_async(cmd, *args, **kwargs):
         return call_command_subprocess(cmd, *args, **kwargs)
 
 
+def call_outside_command_with_output(command, *args, **kwargs):
+    """
+    Runs call_command for a KA Lite installation at the given location,
+    and returns the output.
+    """
+    
+    if settings.IS_SOURCE:
+        assert "kalite_dir" in kwargs, "don't forget to specify the kalite_dir"
+        kalite_dir = kwargs.pop('kalite_dir')
+    else:
+        kalite_dir = None
+    
+    # some custom variables that have to be put inside kwargs
+    # or else will mess up the way the command is called
+    output_to_stdout = kwargs.pop('output_to_stdout', False)
+    output_to_stderr = kwargs.pop('output_to_stderr', False)
+    wait = kwargs.pop('wait', True)
+
+    # build the command
+    if kalite_dir:
+        kalite_bin = os.path.join(kalite_dir, "bin", "kalite")
+    else:
+        kalite_bin = 'kalite'
+
+    cmd = (kalite_bin, "manage", command) if os.name != "nt" else (sys.executable, kalite_bin, "manage", command)
+    for arg in args:
+        cmd += (arg,)
+
+    kwargs_keys = kwargs.keys()
+    
+    # Ensure --settings occurs first, as otherwise docopt parsing barfs
+    kwargs_keys = sorted(kwargs_keys, cmp=lambda x,y: -1 if x=="settings" else 0)
+    
+    for key in kwargs_keys:
+        val = kwargs[key]
+        key = key.replace(u"_",u"-")
+        prefix = u"--" if command != "runcherrypyserver" else u""  # hack, but ... whatever!
+        if isinstance(val, bool):
+            cmd += (u"%s%s" % (prefix, key),)
+        else:
+            # TODO(jamalex): remove this replacement, after #4066 is fixed:
+            # https://github.com/learningequality/ka-lite/issues/4066
+            cleaned_val = unicode(val).replace(" ", "")
+            cmd += (u"%s%s=%s" % (prefix, key, cleaned_val),)
+
+    # we also need to change the environment to point to the the local
+    # kalite settings. This is especially important for when the
+    # central server calls this function, as if we don't change this,
+    # kalitectl.py wil look for centralserver.settings instead of
+    # kalite.settings.
+    new_env = os.environ.copy()
+    new_env["DJANGO_SETTINGS_MODULE"] = kwargs.get("settings") or "kalite.settings"
```

Comment:
This should ideally be configurable and only switched on in the central server. The whole function is only used in a test, though, `centralserver/central/tests/utils/distributed_server_factory.py`.

Reply:
Since all this code is used from the central server's test suite, I'm moving it there.